test/stelace.spec.js

```diff
@@ -457,3 +458,60 @@ test('Methods return array from list endpoints without pagination', (t) => {
       throw err
     })
 })
+
+test('Emits an event when the user session is expired', (t) => {
+  const stelace = getStelaceStub({ keyType: 'pk' })
+
+  stelace.startStub()
+
+  const clock = sinon.useFakeTimers()
+
+  const baseURL = stelace.auth.getBaseURL()
+  const accessToken = encodeJwtToken({ userId: 'user_1' }, { expiresIn: '1h' })
+
+  let called = 0
+
+  stelace.onError('userSessionExpired', () => {
+    const tokenStore = stelace.getTokenStore()
+    tokenStore.removeTokens() // clear tokens for other tests
+
+    called += 1
+  })
+
+  stelace.stubRequest(`${baseURL}/auth/token`, {
+    status: 403,
+    headers: {
+      'x-request-id': 'f1f25173-32a5-48da-aa2f-0079568abea0'
+    },
+    response: {
+      message: 'Refresh token expired'
+    }
+  })
+
+  stelace.stubRequest(`${baseURL}/auth/login`, {
+    status: 200,
+    headers: { 'x-request-id': 'f1f25173-32a5-48da-aa2f-0079568abea0' },
+    response: {
+      tokenType: 'Bearer',
+      userId: 'user_1',
+      accessToken,
+      refreshToken: '39ac0373-e457-4f7a-970f-20dc7d97e0d4'
+    }
+  })
+
+  return stelace.auth.login({ username: 'foo', password: 'secretPassword' })
+    .then(() => {
+      clock.tick(3601 * 1000)
+      return stelace.assets.list()
+    })
+    .then(() => {
+      stelace.stopStub()
+      clock.restore()
+      throw new Error('Should not execute this')
+    })
+    .catch(() => {
+      t.is(called, 1)
```

Comment:
May be you should test same request succeeds before ticking the clock (-> 2 calls).

Reply:
Right, I'll do it.